vm/src/bytesinner.rs

```diff
@@ -557,35 +557,46 @@ impl PyBytesInner {
     }
 
     pub fn fromhex(string: &str, vm: &VirtualMachine) -> PyResult<Vec<u8>> {
-        // first check for invalid character
-        for (i, c) in string.char_indices() {
-            if !c.is_digit(16) && !c.is_whitespace() {
-                return Err(vm.new_value_error(format!(
-                    "non-hexadecimal number found in fromhex() arg at position {}",
-                    i
-                )));
+        let mut iter = string.bytes().enumerate();
+        let mut bytes: Vec<u8> = Vec::with_capacity(string.len() / 2);
+        let i = loop {
+            let (i, b) = match iter.next() {
+                Some(val) => val,
+                None => {
+                    return Ok(bytes);
+                }
+            };
+
+            if is_py_ascii_whitespace(b) {
+                continue;
             }
-        }
 
-        // strip white spaces
-        let stripped = string.split_whitespace().collect::<String>();
+            let top = match b {
+                b'0'..=b'9' => b - b'0',
+                b'a'..=b'f' => 10 + b - b'a',
+                b'A'..=b'F' => 10 + b - b'A',
+                _ => break i,
+            };
 
-        // Hex is evaluated on 2 digits
-        if stripped.len() % 2 != 0 {
-            return Err(vm.new_value_error(format!(
-                "non-hexadecimal number found in fromhex() arg at position {}",
-                stripped.len() - 1
-            )));
-        }
+            let (i, b) = match iter.next() {
+                Some(val) => val,
+                None => break i,
```

Comment:
```suggestion
                None => break i + 1,
```
Looks like `bytes.fromhex('7')` throws `ValueError: non-hexadecimal number found in fromhex() arg at position 1`

Reply:
You are right.